File: Chess/ChessServerConsolePy/Node.py
from Pawn import *

from DynamicArray import *

class Node:
    CurrentLocalPawnList =DynamicArray()
    ParentIndexInCurrentTree=-1
    Level=0
    Weight=0
    Location=""
    OldPositionName=""
    Colore =""
    ChildList=DynamicArray()
    BestChildPosition=""
    Parent=object()
    Parent=None
    AssociatePawn=object()

    def GetCurrentLocalPawnListAllier(self):
        resultList =  DynamicArray()
        c =0
        while c < len(self.CurrentLocalPawnList):
            if(self.CurrentLocalPawnList[c].Colore == self.Colore):
                resultList.append(self.CurrentLocalPawnList[c])
            c = c+1
        return resultList

    def __init__(self):
        return

    def __init__(self,currentLocalPawnList):
        self.CurrentLocalPawnList = currentLocalPawnList

File: Chess/ChessBackup13.08.56-25.06.2021/Chess/Utils/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Chess.Utils
{
  public class Node
  {

    public List<Pawn> CurrentLocalPawnList { get; set; }
    public int Level { get; set; }
    public int MakeCheckmateLevel { get; set; }
    public int Weight { get; set; }
    public string Location { get; set; }
    public string OldPositionName { get; set; }

    public string Colore { get; set; }

    public Node Parent { get; set; }
    public List<Node> ChildList { get; set; }

    public string BestChildPosition { get; set; }
    public  Pawn AssociatePawn { get; set; }

    //public TreeNode<CircleNode> ParentTreeNode { get; set; }

    public List<Pawn> GetCurrentLocalPawnList()
    {
      return CurrentLocalPawnList;
    }

    public List<Pawn> GetCurrentLocalPawnListAllier()
    {
      return CurrentLocalPawnList.Where(x=>x.Colore == Colore).ToList();
    }
    public Node()
    {
    }
    public Node(int level, int makeCheckmateLevel, int weight,string  location, string oldPositionName,string colore,Node parent, List<Node> childList,string bestChildPosition,Pawn associatePawn, List<Pawn> currentLocalPawnList)
    {
      Level = level;
      MakeCheckmateLevel = makeCheckmateLevel;
      Weight = weight;
      Location = location;
      OldPositionName = oldPositionName;
      Colore = colore;
      ChildList = childList;
      BestChildPosition = bestChildPosition;
      AssociatePawn = associatePawn;
      CurrentLocalPawnList = currentLocalPawnList;
    }


      public Node(List<Pawn> currentLocalPawnList)
    {
      ChildList = new List<Node>();
      BestChildPosition = "";
      CurrentLocalPawnList = null;
      CurrentLocalPawnList = new List<Pawn>();
      CurrentLocalPawnList.AddRange(currentLocalPawnList);
    }

  }
}
